sdk/ai/Azure.AI.Inference/src/Telemetry/OpenTelemetryScope.cs

```diff
@@ -0,0 +1,255 @@
+            {
+                SetTagMaybe(
+                    GenAiRequestTopPKey,
+                    JsonSerializer.Deserialize<double?>(requestOptions.AdditionalProperties["top_p"]));
+            }
+            if (_traceContent)
+            {
+                // Log all messages as the events.
+                foreach (ChatRequestMessage message in requestOptions.Messages)
+                {
+                    TagList requestTags = new() {
+                    { GenAiSystemKey, GenAiSystemValue},
+                    { GenAiEventContent, GetContent(message)}
+                };
+                    _activity?.AddEvent(
+                        new ActivityEvent(
+                            $"gen_ai.{message.Role}.message",
+                            default,
+                            new ActivityTagsCollection(requestTags)
+                        )
+                    );
+                }
+            }
+            _duration = Stopwatch.StartNew();
+        }
+
+        public void RecordResponse(ChatCompletions response)
+        {
+            RecordResponseInternal(new SingleRecordedResponse(response, _traceContent));
+        }
+
+        public void UpdateStreamResponse<T>(T item)
+        {
+            if (!_enableTelemetry || item is not StreamingChatCompletionsUpdate)
+                return;
+            StreamingChatCompletionsUpdate castedItem = item as StreamingChatCompletionsUpdate;
+            m_recordedStreamingResponse.Update(castedItem);
+        }
+
+        public void RecordStreamingResponse()
+        {
+            RecordResponseInternal(m_recordedStreamingResponse);
+        }
+
+        /// <summary>
+        /// Log the error.
+        /// </summary>
+        /// <param name="e">Exception thrown by completion call.</param>
+        public void RecordError(Exception e)
+        {
+            if (!_enableTelemetry)
+                return;
+            if (!string.IsNullOrEmpty(_caller))
+                m_source.Write(_caller + ".Exception", e);
+            var errorType = e?.GetType()?.FullName;
+            var tags = _commonTags;
+            tags.Add(ErrorTypeKey, errorType);
+            s_duration.Record(_duration.Elapsed.TotalSeconds, tags);
+            if ( e is Azure.RequestFailedException requestFailed)
+            {
+                errorType = requestFailed.Status.ToString();
+            }
+            _activity?.SetTag(ErrorTypeKey, errorType);
+            _activity?.SetStatus(ActivityStatusCode.Error, e?.Message ?? errorType);
+        }
+
+        /// <summary>
+        /// Record the events and metrics associated with the response.
+        /// </summary>
+        /// <param name="recordedResponse"></param>
+        private void RecordResponseInternal(AbstractRecordedResponse recordedResponse)
+        {
+            if (!_enableTelemetry || recordedResponse.IsEmpty)
```

Comment:
I believe we should treat empty response as an error

Reply:
In this scenario, the response can be empty / null only if we have thrown the error during the first SSE event [processing](https://github.com/Azure/azure-sdk-for-net/blob/2632c00b7d36f4f9a53c1718a18975a47a9a92af/sdk/ai/Azure.AI.Inference/src/Helpers/SseAsyncEnumerator.cs#L70C24-L70C47). In this case, the error was already recorded. If we have got the unexpected response from the server, the parser error will be thrown.